### PR TITLE
Use `quickinstall` for installing cargo tools

### PR DESCRIPTION
**Pull request by LesnyRumcajs**
**Summary of changes**
Changes introduced in this pull request:
- this should fix the flaky lint step check plus introduce a faster cargo tool installation when we don't hit the cache. 

Example fiasco: https://github.com/ChainSafe/forest/actions/runs/3574718464/jobs/6012629867

**Reference issue to close (if applicable)**

Closes 

**Other information and links**

### CODE REVIEW

**Review by hanabi1224 — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -13,7 +13,10 @@ install-deps:
 	apt-get install --no-install-recommends -y build-essential clang protobuf-compiler ocl-icd-opencl-dev aria2 cmake
 
 install-lint-tools:
```
Comment (hanabi1224) [inline Makefile:15]: It will be great to also change `Install MDBook` step in `docs.yml`

Comment (LesnyRumcajs) [reply Makefile:15]: Good idea!

File: Makefile [hunk 2]
```diff
@@ -13,7 +13,14 @@ install-deps:
...
-	RUSTFLAGS="-Cstrip=symbols" cargo install --locked taplo-cli cargo-audit cargo-spellcheck cargo-udeps
+	-which taplo && cargo quickinstall taplo-cli
+	-which cargo-audit && cargo quickinstall cargo-audit
```
Comment (hanabi1224) [inline Makefile:17]: Should this be || ?

Comment (LesnyRumcajs) [reply Makefile:17]: Oh well, I'll fix this and not write a single line of code today because my brain got a Guru Meditation

**Review by elmattic — APPROVED** [review]
Thanks!

**Pull request merged**